### PR TITLE
ENT-4720: Add internal endpoint for syncing organization by id

### PR DESCRIPTION
**Pull request by kflahert**
https://issues.redhat.com/browse/ENT-4720

Test Steps:

- Run capacity-ingress
- `curl -X 'PUT'   'http://localhost:8000/api/rhsm-subscriptions/v1/internal/subscriptions/sync/org/123' -H 'x-rh-swatch-psk: YzlhOTg3NTMtMjA5Mi00NjE3LWIyMjYtNWMyNjUzMzMwYjNk'`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -115,6 +115,7 @@ project(":api") {
 
     ext {
         api_spec_path = "${projectDir}/rhsm-subscriptions-api-spec.yaml"
+        internal_api_spec_path = "${projectDir}/internal-rhsm-subscriptions-api-spec.yaml"
```
Comment (kahowell) [inline build.gradle:118]: See above comment about it being an API specific to subscription syncing.

**Review by kahowell — CHANGES_REQUESTED** [review]

File: api/internal-rhsm-subscriptions-api-spec.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
+openapi: "3.0.2"
```
[the remark this answers is not in the document]
Comment (kflahert) [reply api/internal-rhsm-subscriptions-api-spec.yaml]: updated

File: src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.utilization.api.resources.InternalopenapiJsonApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Serves the OpenAPI spec as /openapi.json. */
+@Component
+public class InternalOpenApiJsonResource implements InternalopenapiJsonApi {
```
Comment (lindseyburnett) [inline src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java:29]: InternalopenapiJsonApi -> InternalOpenApiJsonApi ?

File: src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.utilization.api.resources.InternalopenapiJsonApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Serves the OpenAPI spec as /openapi.json. */
```
Comment (lindseyburnett) [inline src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java:27]: Does it actually serve as /openapi.json or is it /internalopenapi.json?

File: src/main/java/org/candlepin/subscriptions/resource/OpenApiSpecController.java [hunk 5]
```diff
@@ -41,9 +41,15 @@
   @Value("classpath:openapi.yaml")
   private Resource openApiYaml;
 
+  @Value("classpath:internalsubsyncopenapi.yaml")
+  private Resource internalopenApiYaml;
```
Comment (lindseyburnett) [inline src/main/java/org/candlepin/subscriptions/resource/OpenApiSpecController.java:45]: Same comment for these about camel casing the "open"

File: src/main/java/org/candlepin/subscriptions/security/ApiSecurityConfiguration.java [hunk 6]
```diff
@@ -161,7 +165,8 @@ protected void configure(HttpSecurity http) throws Exception {
         .anonymous() // Creates an anonymous user if no header is present at all. Prevents NPEs
         .and()
         .authorizeRequests()
-        .antMatchers("/**/openapi.*", "/**/version", "/api-docs/**", "/webjars/**")
+        .antMatchers(
+            "/**/openapi.*", "/**/internalopenapi.*", "/**/version", "/api-docs/**", "/webjars/**")
```
Comment (lindseyburnett) [inline src/main/java/org/candlepin/subscriptions/security/ApiSecurityConfiguration.java:169]: Thoughts on making a constant at the top of this class named something like "URLS_PERMITTED_WITHOUT_AUTH"?

cc: @kahowell

File: src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java (hunk 4 above)
Comment (kflahert) [reply src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java:27]: Nope this is wrong. I'll fix it to make it internalopenapi.json

File: src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java (hunk 3 above)
Comment (kflahert) [reply src/main/java/org/candlepin/subscriptions/resource/InternalOpenApiJsonResource.java:29]: Makes sense, I'll try changing the generated name with tags.

File: src/main/java/org/candlepin/subscriptions/security/ApiSecurityConfiguration.java (hunk 6 above)
Comment (kflahert) [reply src/main/java/org/candlepin/subscriptions/security/ApiSecurityConfiguration.java:169]: I think that would be cleaner, I'll update.

**Comment by kflahert** [comment]
Something got messed up with my last commit. Looking into it.

**Comment by awood** [comment]
Please hold on merging this.  I want to integrate it with the changes from #822 that hit the develop branch already.

**Review by kahowell — APPROVED** [review]

File: src/main/resources/static/api-docs/internal-subscription-sync.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (kahowell) [reply src/main/resources/static/api-docs/internal-subscription-sync.html:1]: We should look into this. Feel free to add a card on the backlog for this. OTOH, we may not need it as we move more things to quarkus.

**Pull request merged**